### PR TITLE
Add org.kde.plasmatube

### PR DESCRIPTION
**Pull request by plata**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by plata** [comment]
CC @flathub/kde

**Comment by plata** [comment]
bot, build org.kde.plasmatube

**Comment by plata** [comment]
@lnj is this ok for you?

**Comment by plata** [comment]
needs:
- https://github.com/flathub/shared-modules/pull/233
- https://github.com/flathub/shared-modules/pull/232

**Comment by hfiguiere** [comment]
> needs:
> 
>     * [Add python3-yt-dlp shared-modules#233](https://github.com/flathub/shared-modules/pull/233)
> 
>     * [Add libmpv shared-modules#232](https://github.com/flathub/shared-modules/pull/232)

No.

Since they are not in shared-modules they should be in this manifest.

**Comment by plata** [comment]
bot, build org.kde.plasmatube

**Comment by JakobDev** [comment]
You might want to run `flatpak-pip-generator` with the `--checker-data` argument, so you will automatic get a PR when a new Version of yt-dlp is released.

File: org.kde.plasmatube.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            "name": "plasmatube",
+            "buildsystem": "cmake-ninja",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.kde.org/stable/plasma-mobile/23.01.0/plasmatube-23.01.0.tar.xz",
+                    "sha256": "d23d2cb6c669a61f763e48aad44d02fdb141f81e4669ed59954ca9676a501c1a",
+                    "x-checker-data": {
+                        "type": "anitya",
+                        "project-id": 326638,
```
Comment (JakobDev) [inline org.kde.plasmatube.json:32]: This points to to [kasts](https://release-monitoring.org/project/326638/). It looks like Plsmatube is currently not monitored by Anitya.

Comment (espidev) [inline org.kde.plasmatube.json:32]:
```suggestion
                        "project-id": 328147,
```

**Comment by plata** [comment]
@JakobDev I don't know if there was a specific reason to use that yt-dlp version. I don't think so but I would leave this to @lnj.

**Comment by JBBgameich** [comment]
> I don't know if there was a specific reason to use that yt-dlp version. I don't think so but I would leave this to

There isn't, but if you want to stay in sync with the nightly flatpak, you could add something similar to what I am using for audiotube: https://github.com/flathub/org.kde.audiotube/blob/master/update-generated-sources.sh

**Comment by plata** [comment]
bot, build org.kde.plasmatube

**Comment by plata** [comment]
bot, build org.kde.plasmatube

**Comment by plata** [comment]
@espidev @JBBgameich can you answer the open topics in the PR description?

File: org.kde.plasmatube.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "sdk": "org.kde.Sdk",
+    "command": "plasmatube",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--socket=pulseaudio",
+        "--filesystem=home",
```
Comment (JBBgameich) [inline org.kde.plasmatube.json:14]: This shouldn't be necessary

File: org.kde.plasmatube.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        "--talk-name=org.freedesktop.Notifications"
```
Comment (JBBgameich) [inline org.kde.plasmatube.json:15]: This shouldn't be necessary

**Review by JBBgameich — CHANGES_REQUESTED** [review]
There are two unused permissions, after removing these only the minimal set that is required is left.

**Comment by JBBgameich** [comment]
> - [x] All assets referenced in the manifest are redistributable by any party. If not, the unredistributable parts are using an extra-data source type.

It's all under open-source licenses


> - [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 

Not sure who is officially maintaining PlasmTube right now, but personally I'm of course fine with publishing to Flathub

> - [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).

kde.org is owned by KDE, so that's fine.

**Comment by plata** [comment]
bot, build org.kde.plasmatube

**Comment by JBBgameich** [comment]
Can you update the python dependencies to the versions used in the current master branch? It seems like the yt-dlp version used in the stable release doesn't work with youtube any more.

**Comment by plata** [comment]
bot, build org.kde.plasmatube

**Pull request closed without merging**